tests/backup/backup_test.go

```diff
@@ -131,6 +150,121 @@ var _ = Describe("{BackupCreateRestore}", func() {
 	})
 })
 
+// This test performs basic test of starting an application, backing it up and killing stork while
+// performing backup.
+var _ = Describe("{BackupCreateKillStoreRestore}", func() {
+	var contexts []*scheduler.Context
+	var bkpNamespaces []string
+
+	namespaceMapping := make(map[string]string)
+	labelSelectores := make(map[string]string)
+
+	It("has to connect and check the backup setup", func() {
+		Step("Deploy applications", func() {
+			contexts = make([]*scheduler.Context, 0)
+			bkpNamespaces = make([]string, 0)
+			for i := 0; i < Inst().ScaleFactor; i++ {
+				taskName := fmt.Sprintf("backupcreaterestore-%d", i)
+				appContexts := ScheduleApplications(taskName)
+				contexts = append(contexts, appContexts...)
+				for _, ctx := range appContexts {
+					namespace := GetAppNamespace(ctx, taskName)
+					bkpNamespaces = append(bkpNamespaces, namespace)
+					namespaceMapping[namespace] = fmt.Sprintf("%s-restore", namespace)
+				}
+			}
+			ValidateApplications(contexts)
+		})
+
+		// Wait for IO to run
+		time.Sleep(time.Minute * 20)
+
+		Step(fmt.Sprintf("Create Backup [%s]", BackupName), func() {
+			// TODO(stgleb): Add multi-namespace backup when ready in px-backup
+			for _, namespace := range bkpNamespaces {
+				CreateBackup(fmt.Sprintf("%s-%s", BackupName, namespace),
+					SourceClusterName, BLocationName,
+					[]string{namespace}, labelSelectores, orgID)
+			}
+		})
+
+		Step("Kill stork", func() {
+			timeout := time.Duration(rand.Int()%120)*time.Second + 60
+			time.Sleep(time.Duration(timeout) * time.Second)
+
+			t := func(interval sched.Interval) {
+				ctx := &scheduler.Context{
+					App: &spec.AppSpec{
+						SpecList: []interface{}{
+							&appsapi.Deployment{
+								ObjectMeta: meta_v1.ObjectMeta{
+									Name:      storkDeploymentName,
```

Comment:
@vilasd / @stgleb the original intention of the test is to kill  stork leader pod while backup is in progress. The stork deployment will elect new leader and continue remaining backup process. User should also be able to restore such backup successfully. Deleting the entire stork deployment itself will not achieve this purpose. If we delete entire stork deployment then backup itself will discontinue. Correct me if I am wrong.

Reply:
@vinayakshnd I thought this function https://github.com/portworx/torpedo/blob/master/drivers/scheduler/k8s/k8s.go#L1550 will delete only underlying pod of deployment, but not deployment itself. Backup works fine so far.